packages/block-library/src/navigation/placeholder.js

```diff
@@ -285,31 +251,49 @@ function NavigationPlaceholder( { onCreate }, ref ) {
 		}
 
 		const { key } = selectedCreateOption;
+		switch ( key ) {
+			case CREATE_PLACEHOLDER_VALUE:
+				// Do nothing.
+				return;
+
+			case CREATE_EMPTY_OPTION_VALUE: {
+				const blocks = [ createBlock( 'core/navigation-link' ) ];
+				onCreate( blocks );
+				return;
+			}
 
-		if ( key === CREATE_FROM_PAGES_OPTION_VALUE && hasPages ) {
-			const blocks = convertPagesToBlocks( pages );
-			const selectNavigationBlock = true;
-			onCreate( blocks, selectNavigationBlock );
-			return;
-		}
+			case CREATE_FROM_PAGES_OPTION_VALUE: {
+				const blocks = convertPagesToBlocks( pages );
+				const selectNavigationBlock = true;
+				onCreate( blocks, selectNavigationBlock );
+				return;
+			}
 
-		if ( key === CREATE_EMPTY_OPTION_VALUE || ! menuItems?.length ) {
```

Comment:
This caught me out as well when working on this previously, but there's a legit failing test because this `|| ! menuItems?.length` logic is missing. 

When the user selects a menu that has no menu items a default empty navigation-link should be created similar to when an 'empty' nav block is created.

Reply:
Aha, thank you! Fixed and made more explicit in 4d513aa.